docs/notebooks/notdiamond_custom_routing.ipynb

```diff
@@ -0,0 +1,476 @@
+    "\n",
+    "![Create an API key](../docs/guides/integrations/imgs/notdiamond/api-keys.png)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from weave.integrations.notdiamond.custom_router import train_evaluations\n",
+    "\n",
+    "api_key = os.getenv(\"NOTDIAMOND_API_KEY\", \"<YOUR_API_KEY>\")\n",
+    "\n",
+    "preference_id = train_evaluations(\n",
+    "    model_evals=model_train,\n",
+    "    prompt_column=\"prompt\",\n",
+    "    response_column=\"actual\",\n",
+    "    language=\"en\",\n",
+    "    maximize=True,\n",
+    "    api_key=api_key,\n",
+    "    preference_id='28de4887-da50-4d4c-84eb-cd1fa7d22b54'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can then follow the training process for your custom router via the Not Diamond app.\n",
+    "\n",
+    "![Check on router training progress](../docs/guides/integrations/imgs/notdiamond/router-preferences.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once your custom router has finished training, you can use it to route your prompts."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "There are different credentials in the netrc file and the environment. Using the environment value.\n",
+      "/Users/compa/.pyenv/versions/weave-3.11/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
+      "  from .autonotebook import tqdm as notebook_tqdm\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "weave version 0.51.11-dev0 has been retired!  Please upgrade.\n",
+      "Logged in as Weights & Biases user: a9_nd.\n",
+      "View Weave data at https://wandb.ai/notdiamond/dev_testing/weave\n",
+      "🍩 https://wandb.ai/notdiamond/dev_testing/r/call/01924bec-c8a0-7f03-8bf0-4e1730626598\n",
+      "🍩 https://wandb.ai/notdiamond/dev_testing/r/call/01924bec-caf9-7e20-a855-6882d04d9bd6\n",
+      "🍩 https://wandb.ai/notdiamond/dev_testing/r/call/01924bec-cafa-7e70-b534-62dc4dcd642e\n",
+      "🍩 https://wandb.ai/notdiamond/dev_testing/r/call/01924bec-cafb-7872-9e6e-36b1bc572927\n",
+      "🍩 https://wandb.ai/notdiamond/dev_testing/r/call/01924bec-cafb-7872-9e6e-36e172d03844\n",
+      "🍩 https://wandb.ai/notdiamond/dev_testing/r/call/01924bec-d02b-7101-9950-82cb583278d2\n",
+      "Session ID: 4fab123a-0a52-46c5-8302-820f52bf9ac5\n",
+      "Target Model: anthropic/claude-3-5-sonnet-20240620\n"
+     ]
+    }
+   ],
+   "source": [
+    "from notdiamond import NotDiamond\n",
+    "import weave\n",
+    "\n",
+    "weave.init(\"dev_testing\")\n",
```

Comment:
Small nit: you might want to use a specific project name like `notdiamond_tutorial`